### PR TITLE
Fixed issue with mixed devices being used when extracting MACE models on GH200 chips.

### PR DESCRIPTION
**Pull request by G-Seaford**
Hi. When trying to extract a mace model into a json on the GH200 chips, I noticed that torch tensors would be constructed on both cpus and on the gpu, resulting in the model failing. I have added a patch to detect whether cuda is available, and if so,  force tensor generation to use the available device. I have tested this on Isambard-AI and this patch works with no issues, but please let me know if there is anything else that needs adding/changing.

### CODE REVIEW

**Comment by wcwitt** [comment]
Thanks for this! Only time for a quick look now, but any idea why this is not sufficient to force everything to the CPU?

```
map_location=torch.device('cpu'),
```

I have also seen some strange conversion failures recently, so I agree there is a problem.

**Comment by G-Seaford** [comment]
I think you can force everything onto the cpu without issue. I just used my  gpu since I had one that was idle while I ran some cpu calculations. On Isambard-AI we get charged the same amount whether we use a gpu or a cpu, but I doubt there is too much lost from using just the cpu

**Comment by wcwitt** [comment]
Ah, so then is the problem that this works as intended

```
    model = torch.load(
        model,
        map_location=torch.device('cpu'),
        weights_only=False
    ).to(torch.float64)
```
but then subsequent tensors were sometimes created on GPU by default, because we did not specifiy a device? And that caused a mismatch?

**Comment by G-Seaford** [comment]
I believe so, yeah. I’m not sure if the issue emerges when using other hardware for model conversion

**Comment by wcwitt** [comment]
Makes sense. Would you be opposed to setting this always to cpu?
```
device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
```
Among other reasons, that puts everyone in the same code path (including AMD/Intel GPUs).

Also, what's the reason for
```
model.eval()
```

**Comment by G-Seaford** [comment]
Setting ```device = torch.device("cpu")``` sounds good and would make things easier. 

The ```model.eval()``` command switches the torch model into evaluation mode, changing the behaviour of layers like Dropout and BatchNorm, to ensure a deterministic output for a given input without changing the ability for analytical gradients to be taken. Essentially, it ensures that lines like ```Phi = tp(H, Y, R)``` are fully deterministic, and result in the same json file being written for the same input model.  I think it also helps prevent the model itself having states mutated by the the evaluation process.  

 I am not 100% certain it is needed here, since mace does not rely on the Dropout nor BatchNorm blocks, and I think e3nn is suitably well behaved to not need this safeguard.

**Comment by wcwitt** [comment]
Thanks for the contribution!
